### PR TITLE
Feature/create cart

### PR DESCRIPTION
**Pull request by foekinator656**
a working cart!

### CODE REVIEW

**Comment by foekinator656** [comment]
@DragonTamer117

File: src/components/admin/login/login.vue [hunk 1]
```diff
@@ -39,6 +39,7 @@
 <script>
 import axios from "axios";
 import router from "@/router";
+import {StoreVars} from "@/components/shop/shop-item/StoreVars";
```
Comment (DragonTamer117) [inline src/components/admin/login/login.vue]: Needs spaces in between. So like:
`import { StoreVars } from "@/components/shop/shop-item/StoreVars";`

File: src/components/admin/login/login.vue [hunk 2]
```diff
@@ -70,8 +71,10 @@ export default {
 
                 const { data } = await axios.post('http://localhost:8080/api/v1/auth/login', authenticationRequest);
                 this.token = data.token;
```
Comment (DragonTamer117) [inline src/components/admin/login/login.vue]: This is not needed, at line 75 you can just say. `store.token = data.token`

File: src/components/admin/login/login.vue [hunk 3]
```diff
@@ -81,7 +84,7 @@ export default {
                 this.token = "";
```
Comment (DragonTamer117) [inline src/components/admin/login/login.vue]: Here you can say that `store.token = ""`

File: src/components/admin/login/login.vue [hunk 4]
```diff
@@ -70,8 +71,10 @@ export default {
 
                 const { data } = await axios.post('http://localhost:8080/api/v1/auth/login', authenticationRequest);
                 this.token = data.token;
+                const store = StoreVars();
```
Comment (DragonTamer117) [inline src/components/admin/login/login.vue]: I would do this before the try.

File: src/components/shared/order/cart-items/cart-items.vue [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+import { StoreVars } from "@/components/shop/shop-item/StoreVars";
+
+export default defineComponent({
+  name: "cart-items",
+  props: ['storageProduct'],
+  methods: {
+    removeFromOrder(storageProduct) {
+      StoreVars().storageProduct.pop(storageProduct.id);
+      if (this.$parent.$data.totalPrice > 0) {
+        this.$parent.$data.totalPrice = this.$parent.$data.totalPrice - storageProduct.price;
```
Comment (DragonTamer117) [inline src/components/shared/order/cart-items/cart-items.vue]: You can write this as follows:
`this.$parent.$data.totalPrice -= storageProduct.price;`

File: src/components/shared/order/shopping-cart.vue [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+    async getProductsFromOrder() {
+      // let order: Order;
+      const store = StoreVars();
+      this.storageProduct = store.storageProduct;
+      this.calcTotalPrice();
+    },
+    calcTotalPrice() {
+      for (let i = 0; i < this.storageProduct.length; i++) {
+        let product = this.storageProduct[i];
+        this.totalPrice = this.totalPrice + product.price;
```
Comment (DragonTamer117) [inline src/components/shared/order/shopping-cart.vue]: You can change this as follows:
`this.totalPrice += product.price;`

File: src/components/shared/order/shopping-cart.vue [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+    return {
+      storageProduct: [],
+      totalPrice: 0,
+      token: ''
+    }
+  },
+  methods: {
+    async getProductsFromOrder() {
+      // let order: Order;
+      const store = StoreVars();
```
Comment (DragonTamer117) [inline src/components/shared/order/shopping-cart.vue:54]: Would it be possible to use it as a variable in data(). Try to deep yourself into it.

File: src/components/shop/shop-content.vue [hunk 8]
```diff
@@ -19,10 +19,12 @@
 <script>
 import axios from "axios";
 import ShopItem from "@/components/shop/shop-item/shop-item.vue";
+import ShoppingCart from "@/components/shared/order/shopping-cart.vue";
 
 export default {
   name: "shop-content",
-  components: { ShopItem },
+  // eslint-disable-next-line vue/no-unused-components
```
Comment (DragonTamer117) [inline src/components/shop/shop-content.vue]: This is not allowed.

File: src/main.js [hunk 9]
```diff
@@ -1,10 +1,30 @@
 import { createApp } from 'vue';
-
+import { createStore } from 'vuex';
 import App from './App.vue';
+import {createPinia} from "pinia";
```
Comment (DragonTamer117) [inline src/main.js:4]: Change it accordingly to: `import { createPinia } from "pinia";`

File: src/router.js [hunk 10]
```diff
@@ -2,14 +2,15 @@ import { createRouter, createWebHistory } from 'vue-router';
...
 
 const router = createRouter({
     history: createWebHistory(),
     routes: [
         { name: 'shop-content', path: '/', meta: { needsAuth: false }, components: { default: ShopContent } },
         { name: 'login-component', path: '/login', meta: { needsAuth: false }, components: { default: Login } },
-        { name: 'register-component', path: '/register', meta: { needsAuth: false }, components: { default: Register } }
-        // { name: 'shop-item', path: '/payment', meta: { needsAuth: false }, components: { default: ShopItem } },
+        { name: 'register-component', path: '/register', meta: { needsAuth: false }, components: { default: Register } },
+        { name: 'shopping-cart-component', path: '/order', meta: { needsAuth: true }, components: { default: Order } },
```
Comment (DragonTamer117) [inline src/router.js]: What is the authentication you are using?

**Review by DragonTamer117 — CHANGES_REQUESTED** [review]

**Review by DragonTamer117 — CHANGES_REQUESTED** [review]
There a lot of changes that are needed to be done.

File: src/components/shared/order/shopping-cart.vue (hunk 7 above)
Comment (DragonTamer117) [reply src/components/shared/order/shopping-cart.vue:54]: In voice chat we have handled why it's not possible for now.

**Review by DragonTamer117 — APPROVED** [review]
Looks good, except for some weird <p>'s.

**Pull request merged**